### PR TITLE
FM2-313: Add replaces field for ServiceRequest and PriorPrescription for MedicationRequest where they have a previous order

### PR DESCRIPTION
**Pull request by Ruhanga**
…for MedicationRequest where they have a previous order

## Issue I worked on

see https://issues.openmrs.org/browse/FM2-313

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/OrderIdentifierTranslator.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.fhir2.api.translators;
+
+import org.hl7.fhir.r4.model.Identifier;
+import org.openmrs.Order;
+
+public interface OrderIdentifierTranslator {
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/OrderIdentifierTranslator.java:15]: This should really `extend ToFhirTranslator<Order>` or, better yet:

```java
public interface OrderIdentifierTranslator<T extends Order> extends OpenmrsFhirTranslator<T, Identifier>
```

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/MedicationRequestTranslatorImpl.java [hunk 2]
```diff
@@ -79,6 +86,15 @@ public MedicationRequest toFhirResource(@Nonnull DrugOrder drugOrder) {
 		medicationRequest.addReasonCode(conceptTranslator.toFhirResource(drugOrder.getOrderReason()));
 		medicationRequest.addDosageInstruction(dosageTranslator.toFhirResource(drugOrder));
 		
+		if (drugOrder.getPreviousOrder() != null
+		        && (drugOrder.getAction() == Order.Action.DISCONTINUE || drugOrder.getAction() == Order.Action.REVISE)) {
+			medicationRequest.setPriorPrescription(createOrderReference((Order) drugOrder.getPreviousOrder())
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/MedicationRequestTranslatorImpl.java:91]: Isn't the cast here redundant?

**Review by ibacher — COMMENTED** [review]
@Ruhanga Thanks for this! It looks good. Could we also add some coverage for the branches not currently covered for `ServiceRequestTranslatorImpl` and `MedicationRequestTranslatorImpl`?

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/OrderIdentifierTranslator.java (hunk 1 above)
Comment (Ruhanga) [reply api/src/main/java/org/openmrs/module/fhir2/api/translators/OrderIdentifierTranslator.java]: >Could we also add some coverage for the branches not currently covered for ServiceRequestTranslatorImpl and MedicationRequestTranslatorImpl?

Should this be part of the this PR or a new ticket?

Comment (ibacher) [reply api/src/main/java/org/openmrs/module/fhir2/api/translators/OrderIdentifierTranslator.java]: Yeah, actually it looks like the amount of code involved is pretty trivial, so let's hold off on that.

**Comment by ibacher** [comment]
Merged here: https://github.com/openmrs/openmrs-module-fhir2/commit/c3f1387170fb372266bcb695b2833824db960148

**Pull request closed without merging**